arviz/stats/stats.py

```diff
@@ -403,12 +408,10 @@ def loo(data, pointwise=False, reff=None, scale="deviance"):
         raise TypeError("Data must include log_likelihood in sample_stats")
     posterior = inference_data.posterior
     log_likelihood = inference_data.sample_stats.log_likelihood
-    n_chains, n_draws, *shape = log_likelihood.shape
-    n_samples = n_chains * n_draws
-    n_data_points = np.product(shape)
-    dims = [dim for dim in log_likelihood.dims if dim not in ["chain", "draw"]]
-    present_coords = {dim: log_likelihood.coords.indexes[dim] for dim in dims}
-    log_likelihood = log_likelihood.values.reshape(n_samples, n_data_points)
+    log_likelihood = log_likelihood.stack(samples=("chain", "draw"))
+    shape = log_likelihood.shape
+    n_samples = shape[-1]
+    n_data_points = np.product(shape[:-1])
```

Comment:
What if loglikelihood is nD?

Reply:
It still works (i checked locally with the same model as the first plot and the test for this passed too). I decided to not stack the observations dimensions in order to keep this information also in the pointwise loo and waic.

Reply:
I think that the only braking change is the shape of psislw input. Which now, in the 2D (sample, and obs) is transposed from what it used to be. In the case of dataarray inputs, this should not affect (as long as the dimension is calles "samples") but for array inputs it does. It could be changed to the original shape and then call `np.rollaxis` to get the `n_samples` axis to the last position (as expected in xarray apply ufunc).